src/main/java/com/gizasystems/automationexercise/apis/ApisProducts.java

```diff
@@ -29,17 +38,27 @@ public ApisProducts searchProductApi(String searchProductFiled, String searchedP
                 .setParameters(formData, RestActions.ParametersType.FORM)
                 .setContentType(ContentType.URLENC)
                 .setTargetStatusCode(Apis.SUCCESS)
-                .perform();
+                .perform();    
         return this;
     }
 
     //////////////////// Validations \\\\\\\\\\\\\\\\\\\\
+  
+    @Step("Validate All products are listed")
+    public ApisProducts validateAllProductsAreListed(String productName) {
+        api.assertThatResponse()
+                .extractedJsonValue("$.products[(@.length-1)].name")
```

Comment:
This json path needs to be enhanced as it may cause false positive if the order of the json object is changed.

Reply:
yes , it causes an error .. we need to talk about it